app/helpers/jobs_helper.rb

```diff
@@ -1,3 +1,28 @@
 module JobsHelper
 
+	def build_ganglia_link( host, start_seconds, report_type, node_num, size )
+    OODClusters.each do |c|
+      if host.eql?(c.id.to_s)
+        if c.custom_allow?(:ganglia)
+          @ganglia_cluster = OodCluster::Servers::Ganglia.new(c.custom_config(:ganglia))
+          @ganglia_base = @ganglia_cluster.uri.to_s
+          @ganglia_server = @ganglia_cluster.opt_query.fetch(:h, '') % {h: node_num }
+          @ganglia_uri = (@ganglia_base + '&z=' + size + '&cs=' + start_seconds.to_s + '&g=' + report_type+ "&h=" + @ganglia_server).html_safe
+        end
+      end
+    end
+    @ganglia_uri
+  end
+
+  def has_ganglia(host)
+    @flag = false
+    OODClusters.each do |c|
+      if host.eql?(c.id.to_s)
+        if c.custom_allow?(:ganglia)
+          @flag = true
+        end
+      end
+    end
+    @flag
```

Comment:
This method can be replaced with:

```ruby
def has_ganglia(host)
  OODClusters[host].try { |cluster| cluster.custom_allow?(:ganglia)
end
```

Of course if you really want to return true or false instead of true or falsy (false or nil) then do

```ruby
def has_ganglia(host)
  OODClusters[host].try { |cluster| cluster.custom_allow?(:ganglia) || false
end
```

A better approach, however, would be that the Jobstatusdata#cluster return the cluster object reference instead of the cluster id. Then if you were ensured that Jobstatusdata#cluster would always return a cluster instance you could avoid all the guard clauses (like using try) for trying to handle the case that you have nil for a cluster.

Reply:
Also, helper methods should not set instance variables. If you need a local variable, create one instead. (without the @)

Reply:
I have commited and pushed all the changes you suggested. If you want me, I can start working on the better approach you mentioned about Jobstatusdata#cluster as my next project.

Reply:
Hold off on that for now. See comment added about per node erb partial.